melange-build/action.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Build package with melange'
+description: |
+  This action builds a package using Melange, given a config file.
+
+inputs:
+  config:
+    description: |
+      The config file to use for building the package.
+    default: .melange.yaml
+
+  archs:
+    description: |
+      The architectures to use.
+    default: x86_64
+
+  sign-with-temporary-key:
+    description: |
+      Sign packages with a temporary key, useful for multi-stage
+      pipelines.
+    default: false
+
+  signing-key-path:
+    description: |
+      The path for the temporary key if signing is enabled.
+    default: /root/melange.rsa
+
+runs:
+  using: 'composite'
+
+  steps:
+    - uses: chainguard-dev/actions/setup-melange@main
+    # TODO(kaniini): enable keyless signing once it lands
+    - uses: chainguard-dev/actions/melange-keygen@main
```

Comment:
One to think for later:

Does this break with the atomic pattern of the actions in the repo? :thinking: 
What I mean is that perhaps each action could be thought as a separate piece of a puzzle and this is like a combo piece that globs other pieces.

Reply:
Yeah, it should be factored out, which is why I factored the keygen into its own action.  Just need to do a little more work on it, but don't want to break the demo.